test/functional/aws_shared_functions.go

```diff
@@ -320,3 +328,39 @@ func getVpcCidrBlock(session *ec2.EC2, clusterTagName, clusterTagValue string) (
 
 	return aws.StringValue(vpcs[0].CidrBlock), nil
 }
+
+func elasticacheTagsContains(tags []*elasticache.Tag, key, value string) bool {
+	for _, tag := range tags {
+		if *tag.Key == key && *tag.Value == value {
+			return true
+		}
+	}
+	return false
+}
+
+func rdsTagsContains(tags []*rds.Tag, key, value string) bool {
+	for _, tag := range tags {
+		if *tag.Key == key && *tag.Value == value {
+			return true
+		}
+	}
+	return false
+}
+
+func s3TagsContains(tags []*s3.Tag, key, value string) bool {
+	for _, tag := range tags {
+		if *tag.Key == key && *tag.Value == value {
+			return true
+		}
+	}
+	return false
+}
+
+func ec2TagsContains(tags []*ec2.Tag, key, value string) bool {
+	for _, tag := range tags {
+		if *tag.Key == key && *tag.Value == value {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Good candidates for usage of golang generics (creating a single function to handle all cases), but I'm okay with the current approach as well.

Reply:
That was a good idea, I had a quick go and came up with this:
```go
func tagsContains[T *elasticache.Tag | *rds.Tag | *s3.Tag | *ec2.Tag](tags []T, key, value string) bool {
	for _, tag := range tags {
		if *tag.Key == key && *tag.Value == value {
			return true
		}
	}
	return false
}
```

Unfortunately though as the type T is only a pointer to a type you can't access the shared fields of the different type parameters. I think this is just one of the limitations of generics at this time https://tip.golang.org/doc/go1.18.

Reply:
I think it is possible to do it in terms of generics but there will be quite a bit of additional code, unless you have an implementation suggestion

Reply:
I think generics aren't suitable for this function. Another possibility may be to provide an interface instead of type, but I'd say we can keep the existing functions for now, until generics allow access to struct fields.